Bottom: 0c34403eb0c78698b817d2a439eb261c05897d09
Top:    9ed101fe5347328c2760ffed78bc25ea0735f758
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-06-13 14:30:18 +0800

vhost: correctly assign vhost message type

Type of message should not be VHOST_IOTLB_MISS but
VHOST_IOTLB_MSG. Previous code work by chance since their value is
equal.

Fixes:
Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 895eaa2..f74f84e 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -1106,7 +1106,8 @@ ssize_t vhost_chr_read_iter(struct vhost_dev *dev, struct iov_iter *to,
 	if (node) {
 		ret = copy_to_iter(&node->msg, size, to);
 
-		if (ret != size || node->msg.type != VHOST_IOTLB_MISS) {
+		if (ret != size ||
+		    node->msg.iotlb.type != VHOST_IOTLB_MISS) {
 			kfree(node);
 			return ret;
 		}
@@ -1124,7 +1125,7 @@ static int vhost_iotlb_miss(struct vhost_virtqueue *vq, u64 iova, int access)
 	struct vhost_msg_node *node;
 	struct vhost_iotlb_msg *msg;
 
-	node = vhost_new_msg(vq, VHOST_IOTLB_MISS);
+	node = vhost_new_msg(vq, VHOST_IOTLB_MSG);
 	if (!node)
 		return -ENOMEM;
